cifs/parse_cifs.go

```diff
@@ -0,0 +1,228 @@
+		line := scanner.Text()
+		for _, regexpHeader := range regexpHeaders {
+			match := regexpHeader.FindStringSubmatch(line)
+			if 0 == len(match) {
+				continue
+			}
+			for index, name := range regexpHeader.SubexpNames() {
+				if 0 == index || "" == name {
+					continue
+				}
+				value, err := strconv.ParseUint(match[index], 10, 64)
+				if nil != err {
+					continue
+				}
+				stats.Header[name] = value
+			}
+			break
+		}
+		if strings.HasPrefix(line, "Total vfs") {
+			break
+		}
+	}
+	// Parse Shares
+	var tmpSMB1Stats *SMB1Stats
+	var tmpSMB2Stats *SMB2Stats
+	var tmpSessionIDs *SessionIDs
+	// The legacy variable sets the current context. True for SMB1, False for SMB2
+	legacy := true
+	for scanner.Scan() {
+		line := scanner.Text()
+		if legacy {
+			// This part manages the parsing of SMB1
+			for _, regexpSMB1 := range regexpSMB1s {
+				match := regexpSMB1.FindStringSubmatch(line)
+				if 0 == len(match) {
+					// Check for SMB1 Line: "SMBs: 9 Oplocks breaks: 0"
+					// If this Check fails we change to SMB2 Statistics
+					if strings.HasPrefix(line, "SMBs:") && !(strings.Contains(line, "breaks")) {
+						legacy = false
+						tmpSMB2Stats = &SMB2Stats{
+							Stats: make(map[string]map[string]uint64),
+						}
+						stats.SMB2Stats = append(stats.SMB2Stats, tmpSMB2Stats)
+						re := regexp.MustCompile("[0-9]+")
+						find_smb := re.FindAllString(line, 1)
+						tmpSMB2Stats.Stats["smbs"] = make(map[string]uint64)
+						value, err := strconv.ParseUint(find_smb[0], 10, 64)
+						if nil != err {
+							continue
+						}
+						tmpSMB2Stats.Stats["smbs"]["smbs"] = value
+						break
+					}
+					continue
+				}
+				for index, name := range regexpSMB1.SubexpNames() {
+					if 0 == index || "" == name {
+						continue
+					}
+					switch name {
+					case "sessionID":
+						value, err := strconv.ParseUint(match[index], 10, 64)
+						if nil != err {
+							continue
+						}
+						tmpSessionIDs = &SessionIDs{
+							SessionID: value,
+						}
+					case "server":
+						if "" != match[index] {
+							tmpSessionIDs.Server = match[index]
+						}
+					case "share":
+						if "" != match[index] {
+							tmpSessionIDs.Share = match[index]
+						}
+					case "smbs":
+						tmpSMB1Stats = &SMB1Stats{
+							Stats: make(map[string]uint64),
+						}
+						stats.SMB1Stats = append(stats.SMB1Stats, tmpSMB1Stats)
+						value, err := strconv.ParseUint(match[index], 10, 64)
+						if nil != err {
+							continue
+						}
+						tmpSMB1Stats.Stats[name] = value
+					default:
+						value, err := strconv.ParseUint(match[index], 10, 64)
+						if nil != err {
+							continue
+						}
+						if 0 == tmpSMB1Stats.SessionIDs.SessionID {
+							tmpSMB1Stats.SessionIDs.SessionID = tmpSessionIDs.SessionID
+							tmpSMB1Stats.SessionIDs.Server = tmpSessionIDs.Server
+							tmpSMB1Stats.SessionIDs.Share = tmpSessionIDs.Share
+
+						}
+						tmpSMB1Stats.Stats[name] = value
+					}
+				}
+				break
+			}
+		} else {
```

Comment:
In my general experience, the use of "else" in Go means that you're not doing enough early returns.